### ISSUE TITLE
Exported game looks different on two different windows 11 devices (Sun disk and glow doesn't appear on laptop)

### ISSUE THREAD

**Issue by 3DartBlade** [godotengine/godot#84771]
### Godot version

v4.1.3.stable.official [f06b6836a]

### System information

Godot v4.1.3.stable - Windows 10.0.22621 - Vulkan (Forward+) - integrated Intel(R) HD Graphics 620 (Intel Corporation; 31.0.101.2125) - Intel(R) Core(TM) i5-7300U CPU @ 2.60GHz (4 Threads)

### Issue description

**Context:** My friend and I started a project that we're collaborating on via git. I noticed that I do not have a sun disk and glow in the scene view, so we started experimenting. We both exported the .exe where the issue persisted. So he sent me his exported .exe and I tried it out and discovered that I still do not have the sun disk and the glow.
My friend's system info: `Godot v4.1.3.stable - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1080 Ti (NVIDIA; 31.0.15.3742) - Intel(R) Core(TM) i5-10400F CPU @ 2.90GHz (12 Threads)`
We both have the Steam version of the engine so it should be the same (and also in the case of an export it should not matter)

**TL;DR:** The environment looks completely different on different devices in the export.

This concerns us, since this means the final product will not look consistent across different devices.

The only thing I can think of is that Godot has some sort of built-in adaptive performance but I could not find any information on that. If that is the case, pointing us towards a potential solution or customization of this feature would be greatly appreciated.

### Steps to reproduce

1. Create a project
2. Add a Node3D, a camera and a directional light facing opposite to the camera
3. Set camera background to Sky, set Sky to procedural and turn on "glow" on the camera
4. Export the project as a .exe and .pck
5. Run the .exe on both `Godot v4.1.3.stable - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1080 Ti (NVIDIA; 31.0.15.3742) - Intel(R) Core(TM) i5-10400F CPU @ 2.90GHz (12 Threads)` 
and
`Godot v4.1.3.stable - Windows 10.0.22621 - Vulkan (Forward+) - integrated Intel(R) HD Graphics 620 (Intel Corporation; 31.0.101.2125) - Intel(R) Core(TM) i5-7300U CPU @ 2.60GHz (4 Threads)`

### Minimal reproduction project

N/A

**Comment by clayjohn** [comment]
Can you please upload an MRP so that contributors can test on their devices and confirm that they experience the same issue?

From your description it sounds like you may just be configuring your environment differently from your friend.

**Comment by 3DartBlade** [comment]
Of course

### Minimal reproduction project
[Project.zip](https://github.com/godotengine/godot/files/13326587/Project.zip)

### Exported files
[Build.zip](https://github.com/godotengine/godot/files/13326589/Build.zip)

### Screenshots
My view:
![Screenshot 2023-11-11 214938](https://github.com/godotengine/godot/assets/59492834/1bbb9728-a694-4222-a902-acf79e9c3dda)
His view:
![Download](https://github.com/godotengine/godot/assets/59492834/9281c7a7-bb1d-476d-9d20-d42d584d7207)

**Comment by 3DartBlade** [comment]
> From your description it sounds like you may just be configuring your environment differently from your friend.
Git should ensure that does not happen. However, even if that was the case, we both ran the exact same .exe that he has exported so it should not matter what my settings are or even if I have Godot installed or not.

**Comment by 3DartBlade** [comment]
Update:
RX 6700 XT, Ryzen 5 5600X, 32GB Win 11 - produces the one **_with_** the sun disk and glow effect
RX 5700XT, i5 10400F, 16GB Win 10 - produces the one with **_no_** sun disk and glow effect

**Comment by thygrrr** [comment]
![image](https://github.com/godotengine/godot/assets/8904620/148415d0-b05a-4800-9697-00417cba52ca)

`Godot Engine v4.1.3.stable.official.f06b6836a - https://godotengine.org
Vulkan API 1.3.260 - Forward+ - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce RTX 3080`

Are you sure the magenta isn't a shader error (and different drivers may handle the error differently?)

**Comment by thygrrr** [comment]
![image](https://github.com/godotengine/godot/assets/8904620/fe611dff-e0a6-4296-8164-7d4a90b1e712)

I merely turned "glow" on and off on the environment.

**Comment by clayjohn** [comment]
Running the MRP I have the sun disk and glow on my system.

> Godot v4.1.3.stable - Pop!_OS 22.04 LTS - X11 - Vulkan (Forward+) - integrated Intel(R) Xe Graphics (TGL GT2) () - 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz (8 Threads)

In your view it appears that the directional light is disabled completely. Accordingly, I don't think this has anything to do with the Environment. 

Can you do a bit of further debugging, like adding a script that prints whether the DirectionalLight3D in the scene is visible?
